src/pages/dataTableUtilities/getPageInitialiser.js

```diff
@@ -0,0 +1,213 @@
+/**
+ * mSupply Mobile
+ * Sustainable Solutions (NZ) Ltd. 2019
+ */
+
+import { UIDatabase } from '../../database';
+import Settings from '../../settings/MobileAppSettings';
+
+import { newSortDataBy, getAllPrograms } from '../../utilities';
+import { recordKeyExtractor } from './utilities';
+
```

Comment:
There are two other components which use `NewStocktakeBatchModal` and `RegimenDataModal` -- They're very weird components that I hate, but they'll be broken. Not 100% sure the best way to handle. I don't think bother moving their initialisers into the file but keeping it usable in `usePageReducer`?

Reply:
7fe5aed1a656a7173223ed4d6169891948aadd4c.

Updated `NewStocktakeBatchModal` to use new `usePageReducer`. I think `RegimenDataModal` should be fixed in its own issue (it still uses `useReducer` with its own actions and reducer).